File: zul/src/org/zkoss/zul/api/Auxheader.java
/* Auxheader.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 09:27:29     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

import org.zkoss.zk.ui.WrongValueException;

/**
 * An auxiliary header.
 * <p>
 * Default {@link #getZclass}: z-auxheader.(since 3.5.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Auxheader extends org.zkoss.zul.impl.api.HeaderElement {

	/**
	 * Returns number of columns to span this header. Default: 1.
	 */
	public int getColspan();

	/**
	 * Sets the number of columns to span this header.
	 * <p>
	 * It is the same as the colspan attribute of HTML TD tag.
	 */
	public void setColspan(int colspan) throws WrongValueException;

	/**
	 * Returns number of rows to span this header. Default: 1.
	 */
	public int getRowspan();

	/**
	 * Sets the number of rows to span this header.
	 * <p>
	 * It is the same as the rowspan attribute of HTML TD tag.
	 */
	public void setRowspan(int rowspan) throws WrongValueException;

}

File: zul/src/main/resources/web/js/zul/mesh/Auxheader.ts
/* Auxheader.ts

	Purpose:

	Description:

	History:
		Mon May  4 17:00:30     2009, Created by tomyeh

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * An auxiliary header.
 * @defaultValue {@link getZclass}: z-auxheader.
 */
@zk.WrapClass('zul.mesh.Auxheader')
export abstract class Auxheader extends zul.mesh.HeaderWidget {
	/** @internal */
	override _colspan = 1;
	/** @internal */
	override _rowspan = 1;

	/**
	 * @returns number of columns to span this header.
	 * @defaultValue `1`.
	 */
	getColspan(): number {
		return this._colspan;
	}

	/**
	 * Sets the number of columns to span this header.
	 * <p>It is the same as the colspan attribute of HTML TD tag.
	 */
	setColspan(colspan: number, opts?: Record<string, boolean>): this {
		const o = this._colspan;
		this._colspan = colspan;

		if (o !== colspan || opts?.force) {
			var n = this.$n();
			if (n) {
				n.colSpan = colspan;
			}
		}

		return this;
	}

	/**
	 * @returns number of rows to span this header.
	 * @defaultValue `1`.
	 */
	getRowspan(): number {
		return this._rowspan;
	}

	/**
	 * Sets the number of rows to span this header.
	 * <p>It is the same as the rowspan attribute of HTML TD tag.
	 */
	setRowspan(rowspan: number, opts?: Record<string, boolean>): this {
		const o = this._rowspan;
		this._rowspan = rowspan;

		if (o !== rowspan || opts?.force) {
			var n = this.$n();
			if (n) {
				n.rowSpan = rowspan;
			}
		}

		return this;
	}
	
	/** @internal */
	override domAttrs_(no?: zk.DomAttrsOptions): string {
		var /*safe*/ s = super.domAttrs_(no);
		if (this._colspan != 1)
			s += ` colspan="${this._colspan}"`;
		if (this._rowspan != 1)
			s += ` rowspan="${this._rowspan}"`;
		return DOMPurify.sanitize(s);
	}
}